src/main/java/net/fabricmc/loader/impl/discovery/ModDiscoverer.java

```diff
@@ -181,7 +185,7 @@ public List<ModCandidate> discoverMods(FabricLoaderImpl loader, Map<String, Set<
 
 		while ((mod = queue.poll()) != null) {
 			if (mod.getMetadata().loadsInEnvironment(envType)) {
-				if (!ret.add(mod)) continue;
+				if (disabledModIds.contains(mod.getId()) || !ret.add(mod)) continue;
```

Comment:
I think it would be worth logging the disabled mods here? Or maybe above all on one line? I can already imagine support issues where people are wondering why certain mods arent loading and it being impossible to help them by looking at the log file.